packages/react/src/components/IconButton/index.tsx

```diff
@@ -24,20 +23,19 @@ const IconButton = forwardRef(function IconButtonInner<
     size = 'M',
     icon,
     isActive = false,
-    componentAs,
-    as,
+    component,
     ...rest
   }: IconButtonProps<T>,
   ref: ForwardedRef<HTMLButtonElement>
 ) {
   validateIconSize(size, icon)
-  const Component = useMemo(() => as ?? 'button', [as])
+  const Component = useMemo(() => component ?? 'button', [component])
   const classNames = useClassNames('charcoal-icon-button', rest.className)
 
   return (
     <Component
       {...rest}
-      as={componentAs}
+      as={component}
```

Comment:
Since it is no longer an `as` prop, I don't think it is necessary to bind it to html.

Reply:
Thank you, fixed.